src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,287 @@
+        }
+        return optionalCar;
     }
 
     @Override
     public List<Car> getAll() {
-        return Storage.getCars();
+        List<Car> carList = new ArrayList<>();
+        String query = "SELECT id from cars where deleted = false";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(getDriversByIdCars(connection,
+                        resultSet.getObject("id", Long.class)));
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't get all data from DBs", exception);
+        }
+        return carList;
     }
 
     @Override
     public Car update(Car car) {
-        List<Car> dbCar = Storage.getCars();
-        IntStream.range(0, dbCar.size())
-                .filter(i -> dbCar.get(i).equals(car))
-                .findFirst()
-                .ifPresent(i -> dbCar.set(i, car));
+        String query = "UPDATE cars set cars.model = ? where cars.id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                            .prepareStatement(query)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getId());
+
+            removeCarsDrivers(connection, car);
+            insertCarsDrivers(connection, car);
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't update the data - "
+                    + car + "in table of cars or cars_drivers", exception);
+        }
+        createCarsDriversDB(car);
         return car;
     }
 
     @Override
     public boolean delete(Long id) {
-        return Storage.getCars().removeIf(i -> i.getId().equals(id));
+        String queryForCars = "UPDATE cars SET deleted = true where id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statementForCars = connection.prepareStatement(queryForCars)) {
+            statementForCars.setLong(1, id);
+            return statementForCars.executeUpdate() > 0;
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't delete the data by id: "
+                    + id + " into cars", exception);
+        }
     }
 
     public List<Car> getAllByDriver(Long driverId) {
-        return getAll().stream()
-                .filter(car -> car.getDrivers()
-                        .stream()
-                        .anyMatch(driver -> driver.getId().equals(driverId)))
-                .collect(Collectors.toList());
+        List<Car> carList = new ArrayList<>();
+        String query = "SELECT id from drivers where id = ? and deleted = false;";
```

Comment:
here let's use the same approach as in `get()` and `getAll()` methods

but note that the first query in this method (where we fetch data about car and manufacturer) will use 3 joins

Reply:
done